docs: update key management docs for keyring-in-Raft (#24026)
hashicorp/nomad: website/content/docs/operations/key-management.mdx

# Key Management

Nomad servers maintain an encryption keyring used to encrypt [Variables][] and
sign task [workload identities][]. The servers encrypt these data encryption
keys (DEK) and store the wrapped keys in Raft.

The key encryption key (KEK) used to encrypt the DEK is controlled by the
[`keyring`][] provider. When using an external KMS or Vault transit encryption

> providers, the KEK is securely stored outside of Nomad. For the default AEAD

Reviewer: ```suggestion
provider, the KEK is securely stored outside of Nomad. For the default AEAD
```
nit

provider, the KEK is stored in cleartext in Raft.

> 

Reviewer 2: ```suggestion
<Note>
- Default AED provider users should be aware that storing the Data Encryption Key (DEK) and Key Encryption Key (KEK) in cleartext in Raft may expose data in case of a breach.
</Note>
```

Author: Done. Reworded slightly because the DEK is never in cleartext.

re-encrypt all variables with the new key. As each key's variables are encrypted
with the new key, the old key will marked as "deprecated".

## Key Decryption

> When a leader is elected, it creates the keyring if it does not already

Reviewer: ```suggestion
When a leader is elected, the leader creates the keyring if it does not already
```

Adding clarification (even tho it seems redundant)

exist. When a key is added, the new wrapped key material is replicated via

> Raft. As each server replicates the new key, it will start a task to decrypt the

Reviewer: ```suggestion
Raft. As each server replicates the new key, the server starts a task to decrypt the
```
present tense

> key material. Until this task completes, the server will not be able to serve

Reviewer: ```suggestion
key material. Until this task completes, the server is not be able to serve
```

requests that require this key.

## Key Redaction in Raft Snapshots

The default AEAD `keyring` configuration stores the KEK in Raft. Raft snapshots

> will contain the cleartext KEK. The `nomad operator snapshot save` command has a

Reviewer: ```suggestion
contain the cleartext KEK. The `nomad operator snapshot save` command has a
```
